src/core/actions/unitactions/Attack.java

```diff
@@ -22,20 +23,39 @@ public Unit getTarget() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO compute all the attack actions for super.unit.
-        return null;
+
+        LinkedList<Action> attacks = new LinkedList<>();
+
+        if(isFeasible(gs)){
+            Attack a = new Attack(this.unit);
+            a.target = this.target;
+            attacks.add(a);
+
+        }
+        return attacks;
     }
 
     @Override
     public boolean isFeasible(final GameState gs)
     {
-        //TODO check if this action is feasible.
+        // Check if target in range
+        if(target.getCurrentPosition().x < unit.getCurrentPosition().x  + this.unit.RANGE || target.getCurrentPosition().y < unit.getCurrentPosition().y  + this.unit.RANGE)
+            return true;
         return false;
     }
 
     @Override
     public boolean execute(GameState gs) {
```

Comment:
Is this computing the damage as indicated in the fandom page? https://polytopia.fandom.com/wiki/Combat